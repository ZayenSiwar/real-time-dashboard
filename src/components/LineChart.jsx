import { Line } from "react-chartjs-2";
import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend
} from "chart.js";

ChartJS.register(
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend
);

export default function LineChart() {
  const data = {
    labels: ["Lun", "Mar", "Mer", "Jeu", "Ven", "Sam", "Dim"],
    datasets: [
      {
        label: "Utilisateurs actifs",
        data: [12, 19, 3, 5, 2, 3, 15],
        borderColor: "rgb(75, 192, 192)",
        backgroundColor: "rgba(75, 192, 192, 0.5)",
      },
    ],
  };

  const options = {
    responsive: true,
    plugins: {
      legend: { position: "top" },
      title: { display: true, text: "Statistiques hebdomadaires" },
    },
  };

  return <Line data={data} options={options} />;
}
