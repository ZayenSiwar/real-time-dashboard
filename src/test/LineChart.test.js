import React from 'react';
import { render, screen } from '@testing-library/react';
import LineChart from '../components/LineChart';

// Mock Chart.js and react-chartjs-2
jest.mock('react-chartjs-2', () => ({
  Line: ({ data, options }) => (
    <div data-testid="line-chart">
      <div data-testid="chart-data">{JSON.stringify(data)}</div>
      <div data-testid="chart-options">{JSON.stringify(options)}</div>
    </div>
  ),
}));

describe('LineChart Component', () => {
  it('renders without crashing', () => {
    render(<LineChart />);
    expect(screen.getByTestId('line-chart')).toBeInTheDocument();
  });

  it('displays correct chart data structure', () => {
    render(<LineChart />);
    const chartData = JSON.parse(screen.getByTestId('chart-data').textContent);
    
    expect(chartData).toHaveProperty('labels');
    expect(chartData).toHaveProperty('datasets');
    expect(Array.isArray(chartData.labels)).toBe(true);
    expect(Array.isArray(chartData.datasets)).toBe(true);
  });

  it('has correct labels for days of the week', () => {
    render(<LineChart />);
    const chartData = JSON.parse(screen.getByTestId('chart-data').textContent);
    
    expect(chartData.labels).toEqual([
      "Lun", "Mar", "Mer", "Jeu", "Ven", "Sam", "Dim"
    ]);
  });

  it('has correct dataset configuration', () => {
    render(<LineChart />);
    const chartData = JSON.parse(screen.getByTestId('chart-data').textContent);
    
    expect(chartData.datasets[0]).toHaveProperty('label', 'Utilisateurs actifs');
    expect(chartData.datasets[0]).toHaveProperty('data');
    expect(chartData.datasets[0]).toHaveProperty('borderColor', 'rgb(75, 192, 192)');
    expect(chartData.datasets[0]).toHaveProperty('backgroundColor', 'rgba(75, 192, 192, 0.5)');
  });

  it('has correct chart options', () => {
    render(<LineChart />);
    const chartOptions = JSON.parse(screen.getByTestId('chart-options').textContent);
    
    expect(chartOptions).toHaveProperty('responsive', true);
    expect(chartOptions).toHaveProperty('plugins');
    expect(chartOptions.plugins).toHaveProperty('legend');
    expect(chartOptions.plugins).toHaveProperty('title');
    expect(chartOptions.plugins.title).toHaveProperty('text', 'Statistiques hebdomadaires');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<LineChart />);
    expect(asFragment()).toMatchSnapshot();
  });
});
