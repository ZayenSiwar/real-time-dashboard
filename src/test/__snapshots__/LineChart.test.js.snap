// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`LineChart Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="line-chart"
  >
    <div
      data-testid="chart-data"
    >
      {"labels":["Lun","Mar","Mer","Jeu","Ven","Sam","Dim"],"datasets":[{"label":"Utilisateurs actifs","data":[12,19,3,5,2,3,15],"borderColor":"rgb(75, 192, 192)","backgroundColor":"rgba(75, 192, 192, 0.5)"}]}
    </div>
    <div
      data-testid="chart-options"
    >
      {"responsive":true,"plugins":{"legend":{"position":"top"},"title":{"display":true,"text":"Statistiques hebdomadaires"}}}
    </div>
  </div>
</DocumentFragment>
`;
